### PR TITLE
Add Popcnt()

### PR DESCRIPTION
**Pull request by Jille**
goos: linux
goarch: amd64
pkg: github.com/bwesterb/go-and
cpu: 13th Gen Intel(R) Core(TM) i9-13900
BenchmarkPopcnt-32           	   34485	     32383 ns/op	30879.98 MB/s
BenchmarkPopcntGeneric-32    	   17019	     70787 ns/op	14126.98 MB/s

### CODE REVIEW

File: and_amd64.s [hunk 1]
```diff
@@ -145,3 +145,41 @@ loop:
 	SUBQ    $0x00000001, BX
 	JNZ     loop
 	RET
+
+// func popcntAVX2(a *byte, l uint64) int
+// Requires: POPCNT
+TEXT ·popcntAVX2(SB), NOSPLIT, $0-24
```
Comment (bwesterb) [inline and_amd64.s]: not AVX2 is it

**Comment by bwesterb** [comment]
OnesCount64 generates the POPCNTQ instruction. Do you get the same performance by unrolling the generic implementation?

File: and_amd64.s (hunk 1 above)
Comment (Jille) [reply and_amd64.s]: fixed

**Comment by Jille** [comment]
Huh, I'd swear it didn't do that when I checked.

(Not unrolling in ASM was easier than unrolling the Go code) It reduces the performance gains, but it's still an improvement.

```
BenchmarkPopcnt-32           	   20053	     56440 ns/op	17718.01 MB/s
BenchmarkPopcntGeneric-32    	   16987	     70303 ns/op	14224.23 MB/s
PASS
```

**Comment by bwesterb** [comment]
https://godbolt.org/z/5cqf94Yq9

What's the performance if we unroll in plain Go?

**Comment by bwesterb** [comment]
(You can actually do popcnt indirectly in AVX2, but that's a bit trickier, but probably more worthwhile than calling POPCNTQ)

**Comment by bwesterb** [comment]
https://arxiv.org/pdf/1611.07612

**Comment by bwesterb** [comment]
I'll implement it.

**Pull request merged**